### ISSUE TITLE
‘buffer-substring’ is an obsolete generalized variable

### ISSUE THREAD

**Issue by treitmaier** [alphapapa/unpackaged.el#25]
I've been receiving warnings that the `buffer-substring` variable is obsolete in emacs 29.1 RC1.
 
```
~/.emacs.d/.local/straight/build-29.1/unpackaged/unpackaged.el: Warning:
    ‘buffer-substring’ is an obsolete generalized variable.
```

**Comment by alphapapa** [comment]
Yes, and I wish it weren't.  I see no reason to deprecate or obsolete it; it's a very useful Elisp idiom.  I posted about it on emacs-devel, with regard to the deprecation of several such generalized variables, and I was told that I should write an individual defense for each such one that I thought should not be deprecated (seeming to reverse the burden, asking me to defend the status quo and code "in the wild", rather than justifying why certain ones should be deprecated), something which I don't know if I'll find the time to do--who would have time for such a thing, really, when it seems that they were marked as obsolete en masse, on a whim, by one maintainer.

Anyway, one of my other packages also uses such a "deprecated" GV.  I have a long list of issues on my packages, and this particular matter is a very low priority.  But I appreciate your letting me know.
